### PR TITLE
THEMES-986: Convert header nav chain tests

### PR DESCRIPTION
**Pull request by vgalatro**
## Description

This PR converts the header nav chain tests from using Enzyme to using React Testing Library. Some tests were removed because they were redundant, or were testing non-existent functionality.

## Jira Ticket

- [THEMES-986](https://arcpublishing.atlassian.net/browse/THEMES-986)

## Acceptance Criteria

- Convert tests using enzyme to RTL

## Test Steps

1. Checkout this branch `git checkout THEMES-986`
2. Run the block repo's tests: `npm run test`
3. All tests for the header nav chain should pass.

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x] Confirmed this PR has unit test files
  - [x] Ran `npm run test`, made sure all tests are passing
  - [x] If the amount of work to write unit tests for this change are excessive,
        please explain why (so that we can fix it whenever it gets refactored).
- [x] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

[THEMES-986]: https://arcpublishing.atlassian.net/browse/THEMES-986?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by nschubach** [comment]
Haven't done any in depth testing, but if this increases the coverage can we nudge up the acceptance levels in the jest.config.js file?

**Comment by nschubach** [comment]
Blocks current (`npm run test -- --coverage`) is at:
49% statements, 61% branch, 40% function, and 49% lines

After this it goes up to:
53% statements, 66% branch, 44% function, and 54% lines

(prior values were 85% across the board)

**Comment by nschubach** [comment]
image

**Comment by vgalatro** [comment]
@nschubach good call, the threshold values have been updated.

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/nav-widget.jsx [hunk 1]
```diff
@@ -36,20 +36,24 @@ const NavWidget = ({
 				iconRight={<Icon name="HamburgerMenu" />}
 				variant="secondary-reverse"
 				size="small"
+				data-testid="nav-chain-nav-section-button"
 			>
 				{breakpoint === "desktop" ? phrases.t("header-nav-chain-block.sections-button") : null}
 			</Button>
 		) : null);
 
+	// The children prop is not 100% guaranteed to be an array, so this is a bit of future proofing in case something changes.
```
Comment (nschubach) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/nav-widget.jsx]: appreciate the comment and effort, but can we remove it?  (the comment, not the code)

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/queryly-search.jsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ const QuerylySearch = () => {
 
 	return (
 		<Button
-			aria-label={phrases.t("header-nav-chain-block.search-text")}
+			accessibilityLabel={phrases.t("header-nav-chain-block.search-text")}
```
Comment (nschubach) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/_children/queryly-search.jsx:21]: good catch

**Review by nschubach — CHANGES_REQUESTED** [review]
So much of me doesn't like that testing ids get into the production code, but without the babel plugin to strip them I'll pass them for now.

Can we just remove this one comment?

**Review by nschubach — APPROVED** [review]

**Pull request merged**